### PR TITLE
Reportbuilder

### PR DESCRIPTION
**Pull request by danmarsden**
new branch with reportbuilder only changes.

### CODE REVIEW

File: lang/en/reengagement.php [hunk 1]
```diff
@@ -45,6 +47,8 @@
 <a href=https://docs.moodle.org/39/en/Reengagement_activity>More Help</a>';
 
 // Alphabetized.
+
```
Comment (danmarsden) [inline lang/en/reengagement.php:50]: @sumaiyamannan - can you please remove the extra carriage returns here?

File: lang/en/reengagement.php [hunk 2]
```diff
@@ -147,3 +151,13 @@
 $string['privacy:metadata:emailtime'] = 'When this user should be emailed';
 $string['privacy:metadata:emailsent'] = 'Email has been sent';
 $string['privacy:metadata:reengagement_inprogress'] = 'Reengagement activities in progress';
+
+$string['course'] = 'Course ID';
+$string['timecreated'] = 'Time Created';
```
Comment (danmarsden) [inline lang/en/reengagement.php:156]: Moodle uses sentence case, not camel case - so things like "Time Created" should be "Time created" - also "Emailtime" should probably be "Email time"

File: report.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+declare(strict_types=1);
+
+use core_reportbuilder\system_report_factory;
+use mod_reengagement\reengagement_system_report;
+
+
+require_once('../../config.php');
+
+require_login();
+
```
Comment (danmarsden) [inline report.php:34]: missing a require_capability call here - which means that anyone logged in can view this report (security issue) - ideally we should create a new capability - maybe call it "mod/reengagement:viewsitereport" and then check that capability right under the require_login() call.

**Review by danmarsden — COMMENTED** [review]
Thanks @sumaiyamannan - looks like a nice start to the report.

Can you please turn the "courseid" column into a "course" column that shows the Course shortname instead of the id number, and also change the "userid" columns to "user" columns that show the users fullname rather than the id number, also in the "Reengagement name" column can you please make it link directly to the re-engagment view.php page?

thanks!

File: lang/en/reengagement.php (hunk 1 above)
Comment (sumaiyamannan) [reply lang/en/reengagement.php:50]: Resolved. Please refer to the new PR for it.

File: lang/en/reengagement.php (hunk 2 above)
Comment (sumaiyamannan) [reply lang/en/reengagement.php:156]: Resolved. Please refer to the new PR for it.

File: report.php (hunk 3 above)
Comment (sumaiyamannan) [reply report.php:34]: Resolved though I had used a slightly different strategy. Please refer to the new PR for it.

**Comment by danmarsden** [comment]
closing this one as it's been replaced by a newer PR.

**Pull request closed without merging**